### PR TITLE
feat: check license before deploying API in gateway

### PR DESCRIPTION
**Pull request by loriepisicchio**
## Issue

https://gravitee.atlassian.net/browse/APIM-3565

## Description

Check license before deploying API on the gateway

### CODE REVIEW

File: gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/Plan.java [hunk 1]
```diff
@@ -162,6 +162,9 @@ public void setStatus(String status) {
 
     @JsonIgnore
     public List<Plugin> getPlugins() {
-        return this.getFlows().stream().map(Flow::getPlugins).flatMap(List::stream).toList();
+        return Optional
+            .ofNullable(this.getFlows())
```
Comment (jourdiw) [inline gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/Plan.java:166]: How can it be null if the default is an empty list?

File: gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java [hunk 2]
```diff
@@ -76,7 +83,23 @@ private boolean register(ReactableApi<?> api, boolean force) {
         // Get deployed API
         ReactableApi<?> deployedApi = get(api.getId());
 
-        // Does the API have a matching sharding tags ?
+        List<Plugin> plugins;
+        if (api.getDefinitionVersion() == DefinitionVersion.V4) {
```
Comment (jourdiw) [inline gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java:87]: Does this block exist in the 🧅 architecture? If so, could we use it or are we obliged to rewrite it here?

File: gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/Plan.java (hunk 1 above)
Comment (loriepisicchio) [reply gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/Plan.java:166]: It happened in the integration tests, maybe when deserializing definition from file.

File: gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java (hunk 2 above)
Comment (jourdiw) [reply gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java:87]: No onions here. 👍

Comment (loriepisicchio) [reply gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java:87]: This APIManager is in the gateway, so it doesn't access the 🧅 architecture services. This si why we have to repeat this check here

**Review by jourdiw — APPROVED** [review]

**Pull request merged**